Makefile

```
@@ -63,100 +65,105 @@ ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 
-# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy: generate
+deploy: generate ## Deploy controller in the configured Kubernetes cluster in ~/.kube/config
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
-# Generate manifests e.g. CRD, RBAC etc.
-generate:
+generate: ## Generate manifests e.g. CRD, RBAC etc.
 	@CRD_ROOT_DIR="$(CRD_ROOT_DIR)" go generate -tags=generate generate.go
 	@rm config/*.yaml
 
-# Generate CRD to file
-crd: generate
+crd: generate ## Generate CRD to file
 	$(KUSTOMIZE) build $(CRD_ROOT_DIR)/v1 > $(CRD_FILE)
 	$(KUSTOMIZE) build $(CRD_ROOT_DIR)/v1beta1 > $(CRD_FILE_LEGACY)
 
-# Run go fmt against code
-fmt:
+fmt: ## Run go fmt against code
 	go fmt ./...
 
-# Run go vet against code
-vet:
+vet: ## Run go vet against code
 	go vet ./...
 
-lint: fmt vet
+lint: fmt vet ## Invokes the fmt and vet targets
 	@echo 'Check for uncommitted changes ...'
 	git diff --exit-code
 
 # Build the binary without running generators
 $(BIN_FILENAME):
 	$(build_cmd)
 
-# Build the docker image
-docker-build: $(BIN_FILENAME)
+docker-build: $(BIN_FILENAME) $(KIND_KUBECONFIG) ## Build the docker image
 	docker build . -t $(DOCKER_IMG) -t $(QUAY_IMG) -t $(E2E_IMG)
 
-# Push the docker image
-docker-push:
+docker-push: ## Push the docker image
 	docker push $(DOCKER_IMG)
 	docker push $(QUAY_IMG)
 
-# Generate bundle manifests and metadata, then validate generated files.
 .PHONY: bundle
-bundle: generate
+bundle: generate ## Generate bundle manifests and metadata, then validate generated files.
 	operator-sdk generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	operator-sdk bundle validate ./bundle
 
-docs-serve:
+docs-serve: ## Locally run the docs server
 	$(antora_preview_cmd)
 
-install_bats:
+install_bats: ## Installs the bats util via NPM
 	$(MAKE) -C e2e install_bats
 
-e2e_test: docker-build
+e2e_test: install_bats $(SETUP_E2E_TEST) docker-build ## Runs the e2e test suite
 	docker push $(E2E_IMG)
 	$(MAKE) -C e2e run_bats -e KUBECONFIG=../$(KIND_KUBECONFIG)
 
-setup_e2e_test: export KUBECONFIG = $(KIND_KUBECONFIG)
-setup_e2e_test: $(KIND_BIN)
-	@kubectl config use-context kind-$(KIND_CLUSTER)
-	@$(KUSTOMIZE) build $(CRD_ROOT_DIR)/$(CRD_SPEC_VERSION) | kubectl apply $(KIND_KUBECTL_ARGS) -f -
-
 run_kind: export KUBECONFIG = $(KIND_KUBECONFIG)
-run_kind: setup_e2e_test
+run_kind: $(SETUP_E2E_TEST) ## Runs the operator in kind
 	go run ./main.go
 
+.PHONY: setup_e2e_test
+setup_e2e_test: $(SETUP_E2E_TEST) ## Run the e2e setup
+
+.PHONY: clean_e2e_setup
+clean_e2e_setup: export KUBECONFIG = $(KIND_KUBECONFIG)
+clean_e2e_setup: ## Clean the e2e setup (e.g. to rerun the setup_e2e_test)
+	kubectl delete ns k8up-system --wait || true
```

A better command, but `|| true` is still required, in case the kubeconfig is inexistent.
```suggestion
	kubectl delete ns k8up-system --ignore-not-found --force --grace-period=0 || true
```